dlgs_linux.go

```diff
@@ -4,41 +4,53 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/mattn/go-gtk/gtk"
+	"github.com/gotk3/gotk3/glib"
+	"github.com/gotk3/gotk3/gtk"
 )
 
-func init() {
-	gtk.Init(nil)
-}
-
 func closeDialog(dlg *gtk.Dialog) {
 	dlg.Destroy()
-	/* The Destroy call itself isn't enough to remove the dialog from the screen; apparently
-	** that happens once the GTK main loop processes some further events. But if we're
-	** in a non-GTK app the main loop isn't running, so we empty the event queue before
-	** returning from the dialog functions.
-	** Not sure how this interacts with an actual GTK app... */
-	for gtk.EventsPending() {
-		gtk.MainIteration()
```

Comment:
Hm, can you elaborate on why this is no longer needed? Is it something to do with the switch to `glib.IdleAdd`? If that is doing what I think it is doing (submitting a function to be run as part of the main loop), does that break functionality for non-gtk apps?

I tried to test myself but I'm getting build errors upstream and don't have time to investigate myself, eg.:

```
# github.com/gotk3/gotk3/glib
../../gotk3/gotk3/glib/glib.go:260: cannot use idleSrc (type *_Ctype_GSource) as type *_Ctype_struct__GSource in function argument
../../gotk3/gotk3/glib/glib.go:283: cannot use timeoutSrc (type *_Ctype_GSource) as type *_Ctype_struct__GSource in function argument
../../gotk3/gotk3/glib/glib.go:294: cannot use src (type *_Ctype_struct__GSource) as type *_Ctype_GSource in function argument
```

Reply:
I think they are related, yes. I remember we were having problems as we have other gtk parts in our app, and switched to send the dialogs into the main loop.

I see this approach doesn't work properly in programs that doesn't have a gtk main loop already running. I need to investigate more how to support both our needs and the non-gtk app needs.